generator/template_repo.go

```diff
@@ -207,6 +213,58 @@ func asPrettyJSON(data interface{}) (string, error) {
 	return string(b), nil
 }
 
+func goSliceInitializer(data interface{}) (string, error) {
+	// goSliceInitializer constructs a Go literal initializer from interface{} literals.
+	// e.g. []interface{}{"a", "b"} is transformed in {"a","b",}
+	// e.g. map[string]interface{}{ "a": "x", "b": "y"} is transformed in {"a":"x","b":"y",}
+	// NOTE: this is currently used to construct simple slice intializers for default values.
+	// This allows for nicer slice initializers for slices of primitive types and avoid systematic use for json.Unmarshal().
+	b, err := json.Marshal(data)
+	if err != nil {
+		return "", err
+	}
+	return strings.Replace(strings.Replace(strings.Replace(string(b), "}", ",}", -1), "[", "{", -1), "]", ",}", -1), nil
+}
+
+func hasDefaultParam(parameters GenParameters) bool {
+	// hasDefaultParam returns true is at least one input parameter has a .Default
+	// NOTE: this is currently used by templates to avoid empty constructs
+	for _, param := range parameters {
+		if param.HasDefault {
+			return true
+		}
+	}
+	return false
+}
+
+func hasDefaultHeader(headers GenHeaders) bool {
+	// hasDefaultHeader returns true is at least one header in response has a .Default
+	// NOTE: this is currently used by templates to avoid empty constructs
+	for _, header := range headers {
+		if header.HasDefault {
+			return true
+		}
+	}
+	return false
+}
+
+func asItemsLocation(items interface{}) string {
```

Comment:
this is again not something that needs to be done in the templates but rather it's context so should go on the GenXxx models

Reply:
ok. It jast just easier to avoid spreading too much changes everywhere. If now you agree with the change, I can make it to the Gen struct.

Another similar info not mentioned in your comments (because it's stealthier) is the choice of initialization method for default values (i.e. determining if a native initializer is used, a type conversion TYPE() construct, or calling to MarshalText())

This one should be a member of the GenXXX structs as well.

Reply:
OK. 
BTW, this is just to generate a comment in source to more easily relate generated code to spec in nested loops. 
Do you think I should follow a specific guideline to allow some day the "generate spec" feature to benefit from this as well?

Reply:
Done everything here but a "clean" method to select the proper initializer for default values, which is still a shaky hack in templates. Postponed.